src/mixins/helpers.js

```diff
@@ -43,6 +43,8 @@ var helpers = {
     var trackWidth = this.getWidth(ReactDOM.findDOMNode(this.refs.track));
     var slideWidth = this.getWidth(ReactDOM.findDOMNode(this))/props.slidesToShow;
 
+    var currentSlide = props.rtl ? slideCount - 1 - props.initialSlide : props.initialSlide;
```

Comment:
yes,  Issues like #297are happing when slide count change dynamically. I will fix it with the above check I mentioned.

Please open separate PR's when your are fixing multiple issues to make it easy to review